lib/SPIRV/SPIRVLowerConstExpr.cpp

```diff
@@ -162,14 +162,16 @@ void SPIRVLowerConstExpr::visit(Module *M) {
                          [LowerOp](Value *V) { return LowerOp(V); });
           Value *Repl = nullptr;
           unsigned Idx = 0;
+          auto *PhiII = dyn_cast<PHINode>(II);
+          auto *InsPoint = PhiII ? &PhiII->getIncomingBlock(OI)->back() : II;
```

Comment:
What about using [`getFirstInsertionPt`](https://llvm.org/doxygen/classllvm_1_1BasicBlock.html#a161fd4e9fa5367f64c2a4c9e921c3ad3)?

Reply:
AFAIR getFirstInsertionPt return the first instruction in the BB we can insert new inst before, but we can't do that here as phi should use that instructions

Reply:
> AFAIR getFirstInsertionPt returns the first instruction in the BB we can insert new inst before, but we can't do that here as phi already uses that instructions

Ah, got it - we need to insert new instruction into a predecessor BB, so it is already known/computed prior encountering phi - right?